lib/consensus/isaac.go

```diff
@@ -110,31 +119,80 @@ func (is *ISAAC) SelectProposer(blockHeight uint64, roundNumber uint64) string {
 	return is.proposerSelector.Select(blockHeight, roundNumber)
 }
 
+func (is *ISAAC) SaveNodeHeight(b ballot.Ballot) {
+	is.nodesHeight[b.Source()] = b.Round().BlockHeight
+}
+
 func (is *ISAAC) IsAvailableRound(round round.Round) bool {
-	// check current round is from InitRound
-	if is.LatestRound.BlockHash == "" {
-		return true
-	}
+	if round.BlockHeight == is.latestConfirmedBlock.Height {
+		if is.isInitRound(round) {
+			return true
+		}
 
-	if round.BlockHeight < is.latestConfirmedBlock.Height {
-		return false
-	} else if round.BlockHeight == is.latestConfirmedBlock.Height {
 		if round.BlockHash != is.latestConfirmedBlock.Hash {
 			return false
 		}
-	} else {
-		// TODO if incoming round.BlockHeight is bigger than
-		// latestConfirmedBlock.Height and this round confirmed successfully,
-		// this node will get into sync state
+
+		if round.BlockHeight == is.LatestRound.BlockHeight {
+			if round.Number <= is.LatestRound.Number {
+				return false
+			}
+		}
+		return true
+	} else if round.BlockHeight > is.latestConfirmedBlock.Height {
+		is.startSync()
 	}
 
-	if round.BlockHeight == is.LatestRound.BlockHeight {
-		if round.Number <= is.LatestRound.Number {
-			return false
+	return false
+}
+
+func (is *ISAAC) isInitRound(round round.Round) bool {
+	genesisHeight := uint64(1)
+	return is.LatestRound.BlockHash == "" && round.BlockHeight == genesisHeight
+}
```

Comment:
How about `isSync()` and  like `isInitRound`? and
It would be like to keep ISAAC  keep last requested highest block height.
Because that It looks like that this code makes many requests to the syncer.